x-pack/plugins/task_manager/lib/extract_task_definitions.test.ts

```diff
@@ -58,7 +45,7 @@ describe('extractTaskDefinitions', () => {
     const maxWorkers = 10;
     const overrideNumWorkers = {};
     const taskDefinitions = getMockTaskDefinitions({ numTasks: 3 });
-    const result = extractTaskDefinitions(maxWorkers, taskDefinitions, overrideNumWorkers);
+    const result = extractTaskDefinitions(taskDefinitions, maxWorkers, overrideNumWorkers);
```

Comment:
I know I'm the one that named that came up with this name (`extractTaskDefinitions`), but it's not a good name. You're passing it `taskDefinitions`, which would indicate that we've already extracted task definitions from whatever source defined them. What do you think about renaming this to `sanitizeTaskDefinitions` or something?

Also, since we now have an explicit "register" function, maybe that's where the sanitization and validation of of the task definitions belongs? That way, we can throw an error at the time the definition is being registered, rather than later in the cycle.

Reply:
I agree on the renaming. We'll call `sanitizeTaskDefinitions` from the `registerTaskDefinitions` so errors will be thrown at the time a definition is registered. I'll make sure an integration test covers that scenario